pyanaconda/kickstart.py

```diff
@@ -770,51 +770,58 @@ def parse(self, args):
 
         return fc
 
-class Firewall(commands.firewall.F28_Firewall):
+class Firewall(RemovedCommand):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.packages = []
 
+    def __str__(self):
+        # The kickstart for this command is generated by the Firewall sub module
+        return ""
+
     def setup(self):
-        if self.seen:
+        firewall_proxy = NETWORK.get_proxy(FIREWALL)
+        if firewall_proxy.FirewallKickstarted:
             self.packages = ["firewalld"]
 
     def execute(self, storage, ksdata, instClass):
         args = []
 
+        firewall_proxy = NETWORK.get_proxy(FIREWALL)
         # If --use-system-defaults was passed then the user wants
         # whatever was provided by the rpms or ostree to be the
         # default, do nothing.
-        if self.use_system_defaults:
+        if firewall_proxy.UseSystemDefaults:
             firewall_log.info("ks file instructs to use system defaults for "
                               "firewall, skipping configuration.")
             return
 
         # enabled is None if neither --enable or --disable is passed
         # default to enabled if nothing has been set.
-        if self.enabled == False:
```

Comment:
Shouldn't this be `is False`?

Reply:
I have been wondering about that as well - the == comparison is taken verbatim from the code in kickstart.py. And there are slight differences:

    >>> None == False
    False
    >>> None is False
    False
    >>> False is False
    True
    >>> False == False
    True
    >>> 0 == False
    True
    >>> 0 is False
    False

But I guess there should never be 0 used instead of False coming from Pykickstart, so I'll just use is.

Reply:
Btw. `firewall_proxy.FirewallEnabled` will never be None, so `if firewall_proxy.FirewallEnabled` is enough.